nvflare/app_common/workflows/scatter_and_gather_scaffold.py

```diff
@@ -0,0 +1,201 @@
+            aggregator_id=aggregator_id,
+            persistor_id=persistor_id,
+            shareable_generator_id=shareable_generator_id,
+            train_task_name=train_task_name,
+            train_timeout=train_timeout,
+            ignore_result_error=ignore_result_error,
+        )
+
+        # for SCAFFOLD
+        self.aggregator_ctrl = None
+        self._global_ctrl_weights = None
+
+    def start_controller(self, fl_ctx: FLContext) -> None:
+        super().start_controller(fl_ctx=fl_ctx)
+        self.log_info(fl_ctx, "Initializing ScatterAndGatherScaffold workflow.")
+
+        # for SCAFFOLD
+        if not self._global_weights:
+            self.system_panic("Global weights not available!", fl_ctx)
+            return
+
+        self._global_ctrl_weights = copy.deepcopy(self._global_weights["weights"])
+        # Initialize correction term with zeros
+        for k in self._global_ctrl_weights.keys():
+            self._global_ctrl_weights[k] = self._global_ctrl_weights[k] * 0
+        # TODO: Print some stats of the correction magnitudes
+
+    def control_flow(self, abort_signal: Signal, fl_ctx: FLContext) -> None:
+        try:
+
+            self.log_info(fl_ctx, "Beginning ScatterAndGatherScaffold training phase.")
+            self._phase = AppConstants.PHASE_TRAIN
+
+            fl_ctx.set_prop(AppConstants.PHASE, self._phase, private=True, sticky=False)
+            fl_ctx.set_prop(AppConstants.NUM_ROUNDS, self._num_rounds, private=True, sticky=False)
+            self.fire_event(AppEventType.TRAINING_STARTED, fl_ctx)
+
+            for self._current_round in range(self._start_round, self._start_round + self._num_rounds):
+                if self._check_abort_signal(fl_ctx, abort_signal):
+                    return
+
+                self.log_info(fl_ctx, f"Round {self._current_round} started.")
+                fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, self._global_weights, private=True, sticky=True)
+                fl_ctx.set_prop(AppConstants.CURRENT_ROUND, self._current_round, private=True, sticky=False)
+                self.fire_event(AppEventType.ROUND_STARTED, fl_ctx)
+
+                # Create train_task
+                data_shareable: Shareable = self.shareable_gen.learnable_to_shareable(self._global_weights, fl_ctx)
+
+                # add global SCAFFOLD controls using a DXO collection
+                dxo_global_weights = from_shareable(data_shareable)
+                dxo_global_ctrl = DXO(data_kind=DataKind.WEIGHT_DIFF, data=self._global_ctrl_weights)
+                dxo_dict = {
+                    AppConstants.MODEL_WEIGHTS: dxo_global_weights,
+                    AlgorithmConstants.SCAFFOLD_CTRL_GLOBAL: dxo_global_ctrl,
+                }
+                dxo_collection = DXO(data_kind=DataKind.COLLECTION, data=dxo_dict)
+                data_shareable = dxo_collection.update_shareable(data_shareable)  # TODO: add set_dxo() Shareable
```

Comment:
Here you can do: 

data_shareable = dxo_collection.to_shareable()

Reply:
Good point. Changed.